objects/src/accounts/vault.rs

```diff
@@ -0,0 +1,68 @@
+use super::{AccountError, AccountId, Asset, Digest};
+
+// ACCOUNT VAULT
+// ================================================================================================
+
+/// An asset container for an account.
+///
+/// An account vault can contain an unlimited number of assets. The assets are stored in a Sparse
+/// Merkle tree as follows:
+/// - For fungible assets, the index of a node is defined by the issuing faucet ID, and the value
+///   of the node is the asset itself. Thus, for any fungible asset there will be only one node
+///   in the tree.
+/// - For non-fungible assets, the index is defined by the asset itself, and the asset is also
+///   the value of the node.
+///
+/// An account vault can be reduced to a single hash which is the root of the Sparse Merkle tree.
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct AccountVault {
+    // TODO: add backing sparse Merkle tree
+    assets: Vec<Asset>,
+}
+
+impl AccountVault {
```

Comment:
shouldn't we provide also an entrypoint to fetch a merkle opening from a given asset?

Reply:
Probably, and same for `AccountStorage`. But I'm not yet sure how these two fit into a bigger picture (i.e., when we need to read data from a database) - so, skipping some details for now.